tests/actions/IOUTest.ts

```diff
@@ -0,0 +1,3197 @@
+                                        const createdActions = Object.values(reportActionsForIOUReport ?? {}).filter(
+                                            (reportAction) => (reportAction?.actionName as unknown as string) === CONST.REPORT.ACTIONS.TYPE.CREATED,
+                                        );
+                                        const iouActions = Object.values(reportActionsForIOUReport ?? {}).filter(
+                                            (reportAction) => (reportAction?.actionName as unknown as string) === CONST.REPORT.ACTIONS.TYPE.IOU,
+                                        );
+                                        expect(Object.values(createdActions).length).toBe(1);
+                                        expect(Object.values(iouActions).length).toBe(1);
+                                        createdAction = createdActions?.[0] ?? null;
+                                        iouAction = iouActions?.[0] ?? null;
+
+                                        // The CREATED action should not be created after the IOU action
+                                        expect(Date.parse(createdAction?.created ?? '')).toBeLessThan(Date.parse(iouAction?.created ?? ''));
+
+                                        // The IOUReportID should be correct
+                                        expect((iouAction?.originalMessage as IOUMessage)?.IOUReportID).toBe(iouReportID);
+
+                                        // The comment should be included in the IOU action
+                                        expect((iouAction?.originalMessage as IOUMessage)?.comment).toBe(comment);
+
+                                        // The amount in the IOU action should be correct
+                                        expect((iouAction?.originalMessage as IOUMessage).amount).toBe(amount);
+
+                                        // The IOU type should be correct
+                                        expect((iouAction?.originalMessage as IOUMessage).type).toBe(CONST.IOU.REPORT_ACTION_TYPE.CREATE);
+
+                                        // Both actions should be pending
+                                        expect(createdAction?.pendingAction).toBe(CONST.RED_BRICK_ROAD_PENDING_ACTION.ADD);
+                                        expect(iouAction?.pendingAction).toBe(CONST.RED_BRICK_ROAD_PENDING_ACTION.ADD);
+
+                                        resolve();
+                                    },
+                                });
+                            }),
+                    )
+                    .then(
+                        () =>
+                            new Promise<void>((resolve) => {
+                                const connectionID = Onyx.connect({
+                                    key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${transactionThread?.reportID}`,
+                                    waitForCollectionCallback: true,
```

Comment:
It's kind of strange that here `waitForCollectionCallback` is true but the key is not just a collection, but a collection item. It affects typing, cause if `waitForCollectionCallback` is true it expects the whole collection, but we get just one item.